docs/docusaurus/src/theme/NavbarItem/index.js

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import NavbarItem from '@theme-original/NavbarItem';
+import {useNavbarMobileSidebar} from "@docusaurus/theme-common/internal";
+
+export default function NavbarItemWrapper(props) {
+    const mobileSidebar = useNavbarMobileSidebar();
```

Comment:
You can move the condition from the ternary in here so that you always pass the on click
```suggestion
    const mobileSidebar = useNavbarMobileSidebar();
    if (!props.mobile) return
```